scene/gui/code_edit.cpp

```diff
@@ -3139,11 +3139,34 @@ void CodeEdit::_update_scroll_selected_line(float p_mouse_y) {
 	code_completion_force_item_center = -1;
 }
 
+/* Assumes both the new_options and the code_completion_options are sorted. */
+bool CodeEdit::_should_reset_selected_option_for_new_options(const Vector<ScriptLanguage::CodeCompletionOption> &new_options) {
+	if (code_completion_current_selected >= new_options.size())
+		return true;
+
+	for (int i = 0; i < code_completion_options.size() && i < new_options.size(); i++) {
+		/* If we are beyond the currently selected item we don't care about the values later as we are already equal */
+		if (i > code_completion_current_selected) {
+			return false;
+		}
+		/* If we are unequal at an element before the currently selected item we should reset as the elements before the currently selected item have changed. */
+		else if (code_completion_options[i].display != new_options[i].display) {
+			return true;
+		}
+	}
+	/*
+	 * If we are here, then we either exhuasted the smaller of the two lists or checked every option until the currently selected option in both.
+	 * The currently selected option must be inside the new list by here.
+	 * Therefore we should not reset, since the currently selected element is contained in the updated list and nothing up to it has changed.
+	 */
```

Comment:
I feel like this doesn't require such verbose and heavy commenting. The logic is pretty straightforward, and the amount of comments currently rivals the amount of lines of code.